wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -1,31 +1,101 @@
+                type="hidden"
+                name="newsletter_test_email_address"
+                data-wn-panel-target="testAddress"
+            >
+
+            <button
+                type="submit"
+                class="button button-secondary button-longrunning"
+                name="newsletter_action"
+                value="save_campaign"
+                data-controller="w-progress"
+                data-action="w-progress#activate"
+            >
+                {% icon name="spinner" %}
+                Save campaign to {{ backend_name }}
+            </button>
+
+            <button
+                type="button"
+                class="button button-secondary button-longrunning"
+                data-a11y-dialog-show="wn-test-dialog"
+                data-controller="w-progress"
+                data-wn-panel-target="testButton"
+            >
+                {% icon name="spinner" %}
+                Send test email
+            </button>
+
+            {% dialog icon_name="mail" id="wn-test-dialog" title="Send test email" %}
+                <form
+                    data-controller="wn-test"
+                    data-action="wn-test#send:prevent"
+                >
+                    <div class="w-field__wrapper " data-field-wrapper="">
+                        <label class="w-field__label" for="id_text" id="id_text-label">
+                            Email address<span class="w-required-mark">*</span>
+                        </label>
+
+                        <div
+                            class="w-field w-field--char_field w-field--text_input"
+                            data-field=""
+                            data-contentpath="text"
+                        >
+                            <div class="w-field__help" id="id_text-helptext" data-field-help="">
+                                <div class="help">Send a test email to this address</div>
+                            </div>
+
+                            <div class="w-field__input" data-field-input="">
+                                <input
+                                    type="email"
+                                    aria-describedby="id_text-helptext"
+                                    required
+                                    id="id_text"
+                                    data-wn-test-target="address"
+                                    value="{{ user_email }}"
+                                >
+                            </div>
+                        </div>
+                    </div>
+
+                    <button type="submit" class="button button-primary">Send</button>
+                </form>
+            {% enddialog %}
+
+            <button
+                type="submit"
+                class="w-hidden"
+                name="newsletter_action"
+                value="send_test_email"
+                data-wn-panel-target="testSubmit"
+            >
+                Send test email
+            </button>
```

Comment:
This is a hidden submit button, separate from the visible button (that only opens the dialog). Its purpose is to submit the form while including an extra field, `newsletter_action=send_test_email`. We could have used a hidden form input, but if for some reason the form submission got cancelled (e.g. by some other custom logic), that input would have stayed in the form, leading to unexpected results if the user saved the form normally.